ui/app/components/auth/login-form.js

```diff
@@ -4,124 +4,103 @@
-      */
-      if (data?.mfa_requirement) {
-        const parsedMfaAuthResponse = this.auth._parseMfaResponse(data.mfa_requirement);
-        // calls onAuthResponse in parent auth/page.js component
-        this.args.onSuccess(parsedMfaAuthResponse, backendType, data);
-        // return here because mfa-form.js will finish login/authentication flow after mfa validation
-        return;
-      }
+  constructor() {
+    super(...arguments);
+    // todo fetch auth customization config here
+    this.fetchMounts.perform();
+  }
 
-      const authResponse = yield this.auth.authenticate({
-        clusterId,
-        backend: backendType,
-        data,
-        selectedAuth,
-      });
+  get availableMethodTypes() {
+    return supportedTypes(this.version.isEnterprise);
+  }
 
-      // calls onAuthResponse in auth/page.js
-      this.args.onSuccess(authResponse, backendType, data);
-    } catch (e) {
-      if (!this.auth.mfaError) {
-        this.authError = `Authentication failed: ${this.auth.handleError(e)}`;
+  get namespaceInput() {
+    const namespaceQP = this.args.namespace;
+    if (this.flags.hvdManagedNamespaceRoot) {
+      // When managed, the user isn't allowed to edit the prefix `admin/` for their nested namespace
+      const split = namespaceQP.split('/');
+      if (split.length > 1) {
+        split.shift();
+        return `/${split.join('/')}`;
       }
+      return '';
     }
+    return namespaceQP;
   }
 
-  @task
-  @waitFor
-  *pollForOktaNumberChallenge(nonce, mount) {
-    // yield for 1s to wait to see if there is a login error before polling
-    yield timeout(1000);
-    if (this.authError) return;
-
-    this.waitingForOktaNumberChallenge = true;
-    // keep polling /auth/okta/verify/:nonce API every 1s until response returns with correct_number
-    let response = null;
-    while (response === null) {
-      // disable polling for tests otherwise promises reject and acceptance tests fail
-      if (Ember.testing) return;
-
-      yield timeout(1000);
-      response = yield this.auth.getOktaNumberChallengeAnswer(nonce, mount);
+  @action
+  handleAuthSelect(e, idx) {
+    if (idx) {
+      this.selectedAuthMethod = Object.keys(this.authTabs)[idx];
+      this.selectedTabIndex = idx;
+    } else {
+      this.selectedAuthMethod = e.target.value;
     }
-    // display correct number so user can select on personal MFA device
-    this.oktaNumberChallengeAnswer = response;
   }
 
   @action
-  onCancel() {
-    // reset variables and stop polling tasks if canceling login
-    this.authError = null;
-    this.oktaNumberChallengeAnswer = null;
-    this.waitingForOktaNumberChallenge = false;
-    this.authenticate.cancelAll();
-    this.pollForOktaNumberChallenge.cancelAll();
+  handleError(message) {
+    this.errorMessage = message;
   }
+
+  fetchMounts = task(
+    waitFor(async () => {
+      try {
+        const unauthMounts = await this.store.findAll('auth-method', {
```

Comment:
I was getting empty array for this response and that showed the rendered form as empty (screenshot)

I think you'll want to handle that case in this block somewhere

Reply:
<img width="797" alt="Screenshot 2025-03-11 at 11 06 25 AM" src="https://github.com/user-attachments/assets/1ca4d7cb-9ef0-424f-997e-1dff023847e8" />

Reply:
Yes - thank you for pointing out. (And thanks for checking out the branch!!) All of this logic was based on accommodating users who had previously set "listing_visibility" since that was the logic I was most concerned with.

I'll address this though so it's a "working" POC from day 0